File: backend/src/sidra/api/models/rail_unit.py
from __future__ import annotations

from pydantic import BaseModel


class RailUnit(BaseModel):
    """One unit on a track's rail, with the two markers the UI draws on it."""

    ordinal: int
    ref: str

    work_title_en: str
    """The sefer this unit sits in. A track spanning books repeats its addresses in every one of
    them, so the title is what tells two identical-looking positions apart."""

    work_title_he: str

    label_en: str
    label_he: str
    sefaria_url: str | None
    is_actual: bool
    is_scheduled: bool

File: frontend/src/types/RailUnit.ts
export interface RailUnit {
  readonly ordinal: number;
  readonly ref: string;
  /** The sefer the unit sits in. A track spanning books repeats its addresses in each of them. */
  readonly work_title_en: string;
  readonly work_title_he: string;
  readonly label_en: string;
  readonly label_he: string;
  readonly sefaria_url: string | null;
  readonly is_actual: boolean;
  readonly is_scheduled: boolean;
}
